### PR TITLE
feat(no debt certificate): Add delegation

### PR DESCRIPTION
**Pull request by johannaagma**
## What

Add delegation option to No Debt Certificate.
Also add possibility to set feature flag with delegation

## Why

Companies can get their No Debt Certificate

## Checklist:

- [X] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] Formatting passes locally with my changes
- [X] I have rebased against main before asking for a review

### CODE REVIEW

**Review by disaerna — APPROVED** [review]
Code owned file looks good 👍

**Review by alexdiljar — APPROVED** [review]
Júní files look good, feature flag on delegations looking very nice!

**Review by obmagnusson — APPROVED** [review]
LGTM

**Review by sigruntg — APPROVED** [review]

**Review by saevarma — COMMENTED** [review]
Looks good, just two comments regarding the scopes.

File: libs/api/domains/finance/src/lib/api-domains-finance.resolver.ts [hunk 1]
```diff
@@ -222,7 +222,7 @@ export class FinanceResolver {
 
   @Query(() => DebtLessCertificateModel)
   @Audit()
-  @Scopes(ApiScope.financeOverview, ApiScope.internal)
+  @Scopes(ApiScope.noDebtCertificate)
```
[the remark this answers is not in the document]
Comment (johannaagma) [reply libs/api/domains/finance/src/lib/api-domains-finance.resolver.ts:225]: We added this endpoint specifically for the No Debt Certificate:

https://github.com/island-is/island.is/commit/ea994b1c0fefe1348aa47120badeb9b9cd97eee5

Not being used anywhere else, as far as I know

**Review by valurefugl — APPROVED** [review]
Code owned file looks good, might be a merge conflict though?

**Review by saevarma — APPROVED** [review]
LGTM

**Pull request merged**